src/taskmanager.cpp

```diff
@@ -483,6 +519,10 @@ void TaskManager::runScheduledJobs()
            || (doMonthly
                && (job->optionScheduledEnabled() == JobSchedule::Monthly)))
         {
+            // Before the first job, check & wait for an internet connection.
+            if(nothingToDo)
```

Comment:
Why do you wait if there's nothing to do, shouldn't it be the other way around? the waitForOnline check should also be moved outside the loop also and not be conditioned on that bool, if you want you can check the doDaily,Weekly,Monthly bools if desired to not run if there's nothing scheduled. Nothing to do is only used to quit the app immediately, in contrast to deferred quit if a backup is ran indeed.

Reply:
Hmm, I was trying to reuse the existing `nothingToDo` variable name, and ended up with unclear code.  I slightly changed it and added some comments in 9b6587e; is the new code more clear?

If not, I'll try to think of renaming `nothingToDo`.  Or maybe even change it to something like `int jobsScheduled`, accumulate that, and change the checks to `if (jobsScheduled > 0)`.

Thanks for all the comments, @shinnok!